packages/cardano-services/test/Asset/CardanoTokenRegistry.test.ts

```diff
@@ -0,0 +1,74 @@
+import { Asset, Cardano } from '@cardano-sdk/core';
+import { CardanoTokenRegistry } from '../../src/Asset';
+import { InMemoryCache, Key } from '../../src/InMemoryCache';
+import { dummyLogger } from 'ts-log';
+
+describe('CardanoTokenRegistry', () => {
+  const nonvalidAssetId = Cardano.AssetId('0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef');
+  const validAssetId = Cardano.AssetId('f43a62fdc3965df486de8a0d32fe800963589c41b38946602a0dc53541474958');
+
+  describe('return value', () => {
+    const tokenRegistry = new CardanoTokenRegistry({ logger: dummyLogger });
+
+    afterAll(() => tokenRegistry.shutdown());
+
+    it('returns null for nonexisting AssetId', async () => {
+      expect(await tokenRegistry.getTokenMetadata([nonvalidAssetId])).toEqual([null]);
+    });
+
+    it('returns metadata when subject exists', async () => {
+      const [metadata] = await tokenRegistry.getTokenMetadata([validAssetId]);
+
+      expect(metadata).not.toBeNull();
+      if (!metadata) return;
+
+      // Skip icon from test as it is a long long string
```

Comment:
You could instead just assert a value is returned, otherwise, we're not covering the complete API

Reply:
Ok, that also resolve the previous discussion!